### PR TITLE
Improve telemetry data and environment discover metadata 

### PR DESCRIPTION
**Pull request by dingmeng-xue**
## Description

* Enable telemetry on some data plane cmdlet
* Ensure data from generated version and handcrafted version consistent
* Ensure module name is correct
* Adjust discovered environment variable due to compatibility

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Accounts/Accounts/CommonModule/TelemetryProvider.cs [hunk 1]
```diff
@@ -133,49 +137,50 @@ public virtual void Flush()
...
-                }
-                if(!string.IsNullOrEmpty(subscriptionId))
-                {
-                    data.CustomProperties.Add(SubscriptionIdString, subscriptionId);
-                }
+                moduleName = moduleName.Substring(0, moduleName.Length - privateModuleSuffix.Length);
             }
+            qosEvent.ModuleName = moduleName;
+
+
```
Comment (erich-wang) [inline src/Accounts/Accounts/CommonModule/TelemetryProvider.cs:159]: nit: remove unnecessary blank lines

File: src/Accounts/Accounts/CommonModule/TelemetryProvider.cs [hunk 2]
```diff
@@ -133,49 +137,50 @@ public virtual void Flush()
...
+            qosEvent.UserAgent = AzurePSCmdlet.UserAgent;
+            qosEvent.AzVersion = AzurePSCmdlet.AzVersion;
 
             if (invocationInfo != null)
             {
-                data.Parameters = string.Join(",", invocationInfo?.BoundParameters?.Keys?.ToArray());
+                qosEvent.Parameters = string.Join(" ",
```
Comment (erich-wang) [inline src/Accounts/Accounts/CommonModule/TelemetryProvider.cs:165]: change the separator from `,` to ` ` on purpose?

Comment (dingmeng-xue) [reply src/Accounts/Accounts/CommonModule/TelemetryProvider.cs:165]: Yes, it makes generated cmdlet and handcrafted cmdlet have the same pattern
https://github.com/Azure/azure-powershell-common/blob/1a8cbb13e957e26789ec6b327b7b241bc0aaffbc/src/Common/AzurePSCmdlet.cs#L608

**Pull request merged**